bloom-housing/bloom
# fix: add map layer geocoding checks

# Pull Request Template

## Issue Overview

This PR addresses #3814 


## Description

This adds the validation step for all map layer preferences when an application is submitted. It uses the [turfjs pointsWithinPolygon](https://turfjs.org/docs/#pointsWithinPolygon) function to compare the lat long of the submitted application to the feature collection now saved in the mapLayer table.

## How Can This Be Tested/Reviewed?

A sample geojson layer has been provided in the seeded data. It is also the map layer that will be used for the pilot listing when this goes live. 

If you want to see what the layer looks like you can go to this site https://geojson.io/#map=9.32/37.6979/-122.3332 and paste the "featureCollection" portion into it.

Steps to test:
1. In partners - settings page: Create a new preference that has at least one of the option collect address and has the `Yes, check with ArcGIS map` value selected with the appropriate map layer
2. In partners - Listings page: go to a listing, attach that preference to the listing, and click save
3. In public site apply to that listing make sure that you take note of the address you are using or the preference is either inside or outside of the map layer.
4. Go back to partner site and export the application csv for that property
5. The CSV should have the appropriate "yes" or "no" or the application

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
